### PR TITLE
server_status cronjobs: fix empty comments

### PR DESCRIPTION
**Pull request by kate-v-stepanova**
Before it was not displaying comments for some cronjobs. Now it's fixed. Tested locally and on milou-b.

### CODE REVIEW

File: taca/server_status/cronjobs.py [hunk 1]
```diff
@@ -9,35 +9,34 @@
...
-            logging.error(e.message)
-        else:
-            output = p.communicate()[0]
-            crontab = CronTab(tab=output)
+def _parse_crontab():
+    result = {}
+    user = getpass.getuser()
+    logging.info('Getting crontab for user {}'.format(user))
+    try:
+        crontab = CronTab(user=user)
```
Comment (senthil10) [inline taca/server_status/cronjobs.py]: Have tried it in any of the `NAS` ? Cause when I tried (out of curiosity) I got the following error 👀 

```
>>> crontab = CronTab(user='hiseq.bioinfo')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/hiseq.bioinfo/opt/anaconda/envs/master/lib/python2.7/site-packages/crontab.py", line 202, in __init__
    self.read(tabfile)
  File "/home/hiseq.bioinfo/opt/anaconda/envs/master/lib/python2.7/site-packages/crontab.py", line 237, in read
    raise IOError("Read crontab %s: %s" % (self.user, err))
IOError: Read crontab hiseq.bioinfo: must be privileged to use -u
```

Anyway it is not something related to the code itself, maybe we should ask `IT` to fix it (i.e. if it is doable) ?

Comment (Galithil) [inline taca/server_status/cronjobs.py]: I think it's the code actually, it's asking the current user to log in as hiseq.bioinfo and then read the crontab. hiseq.bioinfo can't do that.

Comment (senthil10) [inline taca/server_status/cronjobs.py]: But the current user is `hiseq.bioinfo` 👀  I tried the above as `hiseq.bioinfo`.

I understood the above error like, `crontab` module expects `user` to have certain privilege which `hiseq.bioinfo` don't have at the moment ?

Comment (Galithil) [inline taca/server_status/cronjobs.py]: yeah, that's my guess, I haven't read the crontab module, but the errors means that.

Comment (senthil10) [inline taca/server_status/cronjobs.py]: I just tried it out (as me) and it still gives the same error 😶

File: taca/__init__.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
 """ Main TACA module
 """
 
-__version__ = '0.5.0'
+__version__ = '0.5.1'
```
Comment (senthil10) [inline taca/__init__.py]: @kate-v-stepanova : remember to (pull and) change the version number to right one when you make final push 👍

File: taca/server_status/cronjobs.py [hunk 3]
```diff
@@ -9,35 +9,42 @@
...
+    crontab = None
+    try:
+        crontab = CronTab(user=user)
+    except Exception, e:
+        logging.warning('Cannot get a crontab for user: {}'.format(user))
+        logging.warning(e.message)
         try:
-            p = subprocess.Popen('crontab -l', shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            logging.info('Getting a crontab for default user')
+            crontab = CronTab()
```
Comment (senthil10) [inline taca/server_status/cronjobs.py]: Guess this would return an empty object ? (according to [docs](https://pypi.python.org/pypi/python-crontab))

Comment (kate-v-stepanova) [inline taca/server_status/cronjobs.py]: No, this one returns crontab for current user. 
This was the previous implementation, but for some reasons it doesn't parse comments

Comment (senthil10) [inline taca/server_status/cronjobs.py]: ah, that is strange. Cause when I tried it in `NAS-10` did not work for me

```
>>> from crontab import CronTab
>>> c = CronTab()
>>> c.crons
[]
```

However, `CronTab(user=True)` worked. Maybe difference in `crontab` package version ?

```
>>> c = CronTab(user=True)
>>> c.crons
[<CronJob '@reboot /usr/bin/supervisord -c /home/hiseq.bioinfo/config/supervisord.conf # Ansible: Start supervisord on reboot'>, <CronJob '@hourly /usr/sbin/logrotate /home/hiseq.bioinfo/config/logrotate.conf -s /home/hiseq.bioinfo/config/logrotate.status # Ansible: Run logrotate hourly'>, <CronJob '@daily source ~/.bashrc && taca storage cleanup --site nas --hours 18 # Ansible: Trigger automatic cleanup'>, <CronJob '0 1 * * * source ~/.bashrc && taca backup encrypt # Ansible: Encrypt HiSeq finished runs in nosync'>, <CronJob '*/15 * * * * source /srv/mfs/configs/scripts/check_lsync.sh # Check if the lsync process is running, if not send a mail'>]
```

Anyway, isn't this same as the one in first `try` (`crontab = CronTab(user=user)`) ? As the `user` is got from `getpass.getuser()` (which will also return the current user)

File: taca/__init__.py (hunk 2 above)
Comment (senthil10) [inline taca/__init__.py]: There is a conflict due to this, current [version](https://github.com/SciLifeLab/TACA/blob/master/taca/__init__.py) number is not `0.5.0`

Comment (vezzi) [inline taca/__init__.py]: current version is 0.5.3 (I think)

File: taca/server_status/cronjobs.py (hunk 1 above)
Comment (kate-v-stepanova) [inline taca/server_status/cronjobs.py]: So yes, the problem was in different versions of `python-crontab`, which worked on some NASes and did not on some others. 
@senthil10, can it now be merged?

Comment (senthil10) [inline taca/server_status/cronjobs.py]: @kate-v-stepanova : Nice 👍  and yes it is fine by me. If @vezzi don't have any comments we are good to go.

**Comment by vezzi** [comment]
Looks fine to me

**Comment by senthil10** [comment]
Cool, then will merge it. Great work @kate-v-stepanova 👍

**Pull request merged**